zeroknowledgeproof/src/main/java/com/zeroknowledgeproof/rangeProof/RangeProofProver.kt

```diff
@@ -0,0 +1,199 @@
+package com.zeroknowledgeproof.rangeProof
+
+import java.math.BigInteger
+import java.math.BigInteger.ONE
+import java.math.BigInteger.ZERO
+import java.security.MessageDigest
+import java.security.SecureRandom
+import java.security.Security
+
+class RangeProofProver (private val m: Int, val a: Int, val b: Int, val N: Composite){
+    private val rand = SecureRandom()
+
+    // Security Parameters
+    val t = 80
+    val l = 40
+    val s1 = 40
+    val s2 = 552
+
+    /**
+     * Generate a range proof that [m] is in the range [[a],[b]]
+     * Only the setup part of the rangeproof is performed, still missing is the interactive
+     * verification between prover and verifier.
+     * @param m Int
+     * @param a Int
+     * @param b Int
+     * @return All generated parameters of the proof, excluding
+     */
+    fun rangeSetup(that: RangeProofTrustedParty): Pair<SetupPublicResult, SetupPrivateResult> {
+        // Check whether we can even generate such a proof
+        if (m < a || m > b) {
+            throw ZeroKnowledgeException("Cannot generate a false proof without factoring large " +
+                    "integers \nPlease come back when you can factor large integers.")
+        }
+
+        /////////////////////////////////////////////
+        //Init
+
+        // Generate security parameters
+//        val k1 = BigInteger(2048, rand)
```

Comment:
why is this still unused? can we do the security tuning later? but then why is k2 used?